### PR TITLE
Add ListLinked

### PR DESCRIPTION
**Pull request by Cybernetik**
Début d'implémentation de ListLinked.
Manque les fonctionnalités en commentaires dans le fichier :+1:  :heart:

### CODE REVIEW

File: ListLinked.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+		private Elem head;
+		private int countElem;
+
+		public ListLinked(Object e)
+		{
+			head = new Elem(e);
+		}
+
+		class Elem {
+			private Object info;
```
Comment (Ethiraric) [inline ListLinked.java]: This could be named `held_value` or `value`, since it does not contain information.

File: ListLinked.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	*/
+	class Project {
+		public static void main() {
+			return;
+		}
+	}
+
+	public class ListLinked {
+		private Elem head;
+		private int countElem;
```
Comment (Ethiraric) [inline ListLinked.java]: This should be named `elemCount`. Otherwise it's an element with a count.

File: ListLinked.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
+	/*elem
```
Comment (Ethiraric) [inline ListLinked.java]: Should be marked as `TODO`.

File: ListLinked.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+		{
+			head = new Elem(e);
+		}
+
+		class Elem {
+			private Object info;
+			private Elem next;
+
+			public Elem(Object myInfo)
+			{
```
Comment (Ethiraric) [inline ListLinked.java]: Indent is very inconsistent.

File: ListLinked.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+			countElem++;
+		}
+
+		public void addElemAtPos(Elem e, int pos)
+		{
+		    Elem tmp = head;
+			Elem find;
+			int i = 0;
+			while(i < pos  - 1 && tmp.next != null)
+				{
```
Comment (Ethiraric) [inline ListLinked.java]: This loop and `addElemAtEnd`'s one could be factored into a `getNthElem` method.

File: ListLinked.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+			Elem tmp = head;
+			while(tmp.next != null)
+			{
+				tmp = tmp.next;
+			}
+			tmp.next = new Elem(e);
+			countElem++;
+		}
+
+		public void addElemAtPos(Elem e, int pos)
```
Comment (Ethiraric) [inline ListLinked.java]: You can't add an elem at the first position. `addElemAtPos(/*..*/, 0)` and `addElemAtPos(/*..*/, 1)` do the same thing.

File: ListLinked.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+			Elem tmp = head;
+			while (tmp != to_delete && tmp.next != null) {
+				tmp = tmp.next;
+			}
+			tmp.next = tmp.next.next;
+			countElem--;
+		}
+
+		// GET
+        public Elem getElem(Elem to_find) {
```
Comment (Ethiraric) [inline ListLinked.java]: This should be named `findElem`, since it actually searches.

File: ListLinked.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+            Elem e = head;
+            int i = 0;
+            while (e != to_find) {
```
Comment (Ethiraric) [inline ListLinked.java]: This is a neverending loop if the element is not found in the list.

File: ListLinked.java [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+				}
+			find = tmp.next;
+			tmp.next = new Elem(e);
+			tmp.next.next = find;
+			countElem++;
+		}
+
+		// DEL
+		public void delElemAtPos(int pos)
+		{
```
Comment (Ethiraric) [inline ListLinked.java]: You should check that `pos` is in range.

File: ListLinked.java [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+			tmp.next = tmp.next.next;
+			countElem--;
+		}
+
+		public void delElem(Elem to_delete) {
+			Elem tmp = head;
+			while (tmp != to_delete && tmp.next != null) {
+				tmp = tmp.next;
+			}
+			tmp.next = tmp.next.next;
```
Comment (Ethiraric) [inline ListLinked.java]: You might crash if `tmp.next` is `null`.

File: ListLinked.java (hunk 3 above)
Comment (Cybernetik) [reply ListLinked.java]: :heart:

File: ListLinked.java (hunk 5 above)
Comment (Cybernetik) [reply ListLinked.java]: :+1:

**Pull request closed without merging**